### PR TITLE
ServiceProvider decoupling 

### PR DESCRIPTION
**Pull request by psykokwak4**
Changed documentation structure a little bit.

### CODE REVIEW

File: sdk/cpp/core/CMakeLists.txt [hunk 1]
```diff
@@ -186,6 +190,7 @@ set(spdlog_fmt_bundled_include_location include/spdlog/fmt/bundled)
 set(spdlog_sinks_include_location include/spdlog/sinks)
 
 set(libydk_install_headers
+    src/path_api.hpp
```
Comment (abhikeshav) [inline sdk/cpp/core/CMakeLists.txt]: Can we please keep this list alphabetically sorted?

File: sdk/cpp/core/docsgen/api/namespaces/nmsp_path.rst [hunk 2]
```diff
@@ -2,161 +2,55 @@
...
+===========
 
 Data Structures
 ---------------
 
 .. cpp:namespace:: ydk::path
 
 +--------+----------------------------------------------------------------------------+
-| struct | | :cpp:class:`Annotation <Annotation>`                                     |
+| struct | | :cpp:class:`Annotation<Annotation>`                                      |
```
Comment (abhikeshav) [inline sdk/cpp/core/docsgen/api/namespaces/nmsp_path.rst]: Thanks for fixing these issues!

File: sdk/cpp/core/docsgen/api/path/errors/codec_error.rst [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-=================
+==============
 
 
 .. cpp:namespace:: ydk::path
 
 .. cpp:class:: YCPPCodecError : public YCPPCoreError
 
-Exception that encapsualtes the validation errors for YDK :cpp:class:`CodecService`.
+    Exception that encapsualtes the validation errors for YDK :cpp:class:`CodecService`.
```
Comment (abhikeshav) [inline sdk/cpp/core/docsgen/api/path/errors/codec_error.rst]: Typo: `encapsualtes` --> `encapsulates`

File: sdk/cpp/core/docsgen/api/path/errors/data_validation_error.rst [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
-==========================
+=======================
 
 
 .. cpp:namespace:: ydk::path
 
 .. cpp:class:: YCPPDataValidationError : public YCPPCoreError
 
-Exception that encapsualtes the validation errors on a data tree.
+    Exception that encapsualtes the validation errors on a data tree.
```
Comment (abhikeshav) [inline sdk/cpp/core/docsgen/api/path/errors/data_validation_error.rst]: Typo: `encapsualtes` --> `encapsulates`

File: sdk/cpp/core/docsgen/api/path/errors/path_error.rst [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
-================
+=============
 
 
 .. cpp:namespace:: ydk::path
 
 .. cpp:class:: YCPPPathError : public YCPPCoreError
 
-Exception that encapsualtes the validation errors for YDK Path.
+    Exception that encapsualtes the validation errors for YDK Path.
```
Comment (abhikeshav) [inline sdk/cpp/core/docsgen/api/path/errors/path_error.rst]: Typo: `encapsualtes` --> `encapsulates`

File: sdk/cpp/core/docsgen/api/path/codec.rst [hunk 6]
```diff
@@ -1,7 +1,7 @@
 .. _ref-codecservice:
 
-path::Codec
-===================
```
Comment (abhikeshav) [inline sdk/cpp/core/docsgen/api/path/codec.rst]: Thanks for cleaning this up!

File: sdk/cpp/core/src/service_provider.cpp [hunk 7]
```diff
@@ -25,17 +25,12 @@
...
-#include <string>
-#include <memory>
+#include "path_api.hpp"
 
 namespace ydk {
-class Service
-{
-};
 
+path::ServiceProvider::~ServiceProvider()
```
Comment (abhikeshav) [inline sdk/cpp/core/src/service_provider.cpp]: `ServiceProvider` should be moved out of `path` namespace. This is an important part of this change as we want to decouple `ServiceProvider` completely from path API

`ydk::path::ServiceProvider` --> `ydk::ServiceProvider`

Comment (psykokwak4) [inline sdk/cpp/core/src/service_provider.cpp]: Thanks, will move `ServiceProvider` out of `path` namespace, and address other comments in upcoming commit.

**Pull request merged**